JDBC/OAuthBrowserFlowSampleApp/src/main/java/OAuthHandler.java

```diff
@@ -0,0 +1,134 @@
+            loadStartProperties("properties.example");
+            setupDbForOAuth();
+            loginWithBrowserOAuth();
+        } finally {
+            tearDownForOAuth();
+        }
+    }
+
+    private void loadStartProperties(String filename) throws IOException {
+        Properties props = new Properties();
+        props.load(new FileReader(filename));
+        connectionString = props.getProperty("connectionString");
+        adminUser = props.getProperty("adminUser");
+        adminUserPassword = props.getProperty("adminUserPassword");
+        oAuthUser = props.getProperty("oAuthUser");
+        authName = props.getProperty("authName");
+        clientId = props.getProperty("clientId");
+        clientSecret = props.getProperty("clientSecret");
+        httpsDiscoveryUrl = props.getProperty("httpsDiscoveryUrl");
+    }
+
+    // Login as admin and setup OAuth user and authentication
+    private void setupDbForOAuth() throws SQLException {
+        adminConnection = getAdminConnection();
+        Statement stmt = adminConnection.createStatement();
+        
+        // Array of SQL commands
+        String[] sqlCommands = {
+            "DROP USER IF EXISTS " + oAuthUser + " CASCADE",
+            "DROP AUTHENTICATION IF EXISTS " + authName + " CASCADE",
+            "CREATE USER " + oAuthUser,
+            "CREATE AUTHENTICATION " + authName + " METHOD 'oauth' HOST '0.0.0.0/0'",
+            "GRANT AUTHENTICATION " + authName + " TO " + oAuthUser,
+            "GRANT ALL ON SCHEMA PUBLIC TO " + oAuthUser,
+            "ALTER AUTHENTICATION " + authName + " SET client_id = '" + clientId + "'",
+            "ALTER AUTHENTICATION " + authName + " SET client_secret = '" + clientSecret + "'",
+            "ALTER AUTHENTICATION " + authName + " SET discovery_url = '" + httpsDiscoveryUrl + "'"
+        };
+
+        // Execute each SQL command in the array
+        for (String sqlCommand : sqlCommands) {
+            stmt.execute(sqlCommand);
+        }
+    }
+    
+    // password auth for admin user
+    private Connection getAdminConnection() throws SQLException {
+        Properties jdbcOptions = new Properties();
+        jdbcOptions.put("user", adminUser);
+        jdbcOptions.put("password", adminUserPassword);
+        Connection conn = DriverManager.getConnection(connectionString, jdbcOptions);
+        return conn;
+    }
+
+    // connect and run SELECT CURRENT_USER to verify browser OAuth succeeded
+    public void loginWithBrowserOAuth() throws SQLException, IOException {
+        String jsonConfigFromFile = loadJsonConfig("OAuthJsonConfig.json");
+        Properties jdbcOptions = new Properties();
+        jdbcOptions.put("authmethod", "browser");
+        jdbcOptions.put("oauthjsonconfig", jsonConfigFromFile);
+        
+        Connection conn = DriverManager.getConnection(connectionString, jdbcOptions);
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("SELECT CURRENT_USER;");
```

Comment:
Maybe our sample query should be one that looks to see the authentication method that was used to log in that user. To show that we are using the right auth method. Something like
`select user_name, authentication_method from sessions where session_id = current_session();`